distributed/_concurrent_futures_thread.py

```diff
@@ -50,6 +50,28 @@ def _python_exit():
 atexit.register(_python_exit)
 
 
+class WorkerThreadInterrupt(threading.ThreadError):
+    pass
+
+
+def _async_raise(tid: int, exctype: type = WorkerThreadInterrupt):
+    """raise exception in given thread"""
+    import ctypes
+
+    if not issubclass(exctype, BaseException):
+        raise TypeError("Must pass an exception type")
+    res = ctypes.pythonapi.PyThreadState_SetAsyncExc(
+        ctypes.c_long(tid), ctypes.py_object(exctype)
+    )
+    if res == 0:
+        raise threading.ThreadError("No thread got set")
+    elif res != 1:  # pragma: no cover
+        # if res>1, somehow more than one thread was set, so immediately roll
+        # back and raise. Should never happen.
+        ctypes.pythonapi.PyThreadState_SetAsyncExc(tid, 0)
+        raise threading.ThreadError("PyThreadState_SetAsyncExc failed")
```

Comment:
I don't think we actually need this block. The docs specifically mention zero and one. res>1 is not defined and it is unclear to me what we should even do in this case, let alone how to "roll back" that situation.

Even peaking into the implementation, this case is not possible, see https://github.com/python/cpython/blob/7f61d9d84843e3445f62eb00c47902f0daa30a72/Python/pystate.c#L1165-L1202